src/rng.h

```diff
@@ -7,6 +7,9 @@
 #include <functional>
 
 long rng( long val1, long val2 );
+class time_duration;
+// implemented in calendar.cpp
+time_duration rng( time_duration val1, time_duration val2 );
```

Comment:
Just move the declaration to calendar.h too, no need to rng to know about this type.